core/src/com/fallenflame/game/GameEngine.java

```diff
@@ -445,5 +453,154 @@ public void setScreenListener(ScreenListener listener) {
         this.listener = listener;
     }
 
+    /************************ POLLING INPUT HANDLING ************************/
+
+    /**
+     * Get mouse position
+     * @return Vector2 mouse position
+     */
+    public Vector2 getMousePosition(){
+        return new Vector2(Gdx.input.getX(), Gdx.input.getY());
+    }
+
+    /**
+     * Reads the input for the player and converts the result into game logic.
+     */
+    public void readInput() {
+        // Copy state from last animation frame
+        // Helps us ignore buttons that are held down
+        resetPrevious  = resetPressed;
+        debugPrevious  = debugPressed;
+        debug2Previous = debug2Pressed;
+        exitPrevious = exitPressed;
+        flarePrevious = flarePressed;
+        sprintPrevious = sprintPressed;
+        sneakPrevious = sneakPressed;
+
+        readKeyboard();
+    }
+
+    /**
+     * Reads input from the keyboard.
+     *
+     * This controller reads from the keyboard regardless of whether or not an X-Box
+     * controller is connected.  However, if a controller is connected, this method
+     * gives priority to the X-Box controller.
+     *
+     */
+    private void readKeyboard() {
+        // Give priority to gamepad results
+        resetPressed = (Gdx.input.isKeyPressed(Input.Keys.R));
+        debugPressed = (Gdx.input.isKeyPressed(Input.Keys.G));
+        debug2Pressed = (Gdx.input.isKeyPressed(Input.Keys.E));
+        exitPressed  = (Gdx.input.isKeyPressed(Input.Keys.ESCAPE));
+        flarePressed  = (Gdx.input.isButtonPressed(Input.Buttons.LEFT));
+        sprintPressed = (Gdx.input.isKeyPressed(Input.Keys.SHIFT_LEFT));
+        sneakPressed = (Gdx.input.isKeyPressed(Input.Keys.CONTROL_LEFT));
+
+        // Directional controls
+        horizontal = 0.0f;
+        if (Gdx.input.isKeyPressed(Input.Keys.D)) {
+            horizontal += 1.0f;
+        }
+        if (Gdx.input.isKeyPressed(Input.Keys.A)) {
+            horizontal -= 1.0f;
+        }
+
+        // TODO: REMOVE CODE BELOW WHEN MOUSE WHEEL IS FIXED --> We should still include this as a functional alternative if possible
+        //#region mouse wheel alternative
+        if(Gdx.input.isKeyPressed(Input.Keys.PERIOD)){
+            lightRadius += 1.0f;
```

Comment:
This change is something that needed to be done anyways. I think this should changed to level.lightFromPlayer(1.0f), since right now this doesn't do anything because we don't use lightRadius at all.